### PR TITLE
feat(combo-box): external input ref support

### PR DESCRIPTION
**Pull request by AlanFnz**
## What

This PR adds support for an external `ref` in the `ComboBox` component, enabling parent components to pass their own refs for access to the input element within the `ComboBoxInput` 

### Media

https://github.com/user-attachments/assets/18122ae2-ffbf-48ca-b317-5dbd1fa4db18

## Why

So we can programmatically focus or interact with the input field within the `ComboBox`
​
A requirement of UI improvement SUP-45087

### CODE REVIEW

File: src/utils/hooks/useMergeRefs.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+export default function useMergeRefs<T>(
+    ...inputRefs: (React.Ref<T> | undefined)[]
+): React.Ref<T> | React.RefCallback<T> {
+    const filteredInputRefs = inputRefs.filter(Boolean);
```
Comment (lloydaf) [inline src/utils/hooks/useMergeRefs.ts:4]: Is this the same as `inputRefs.filter(ref => ref)`?

File: src/components/experimental/ComboBox/docs/ComboBox.stories.tsx [hunk 2]
```diff
@@ -153,3 +154,81 @@ export const FullyControlled: Story = {
...
+            return () => {
+                ignore = true;
+            };
+        }, [filterText]);
+
+        const handleFocus = () => {
+            if (comboBoxRef.current) {
+                comboBoxRef.current.focus();
+                // eslint-disable-next-line no-console
+                console.log('input focused');
```
Comment (lloydaf) [inline src/components/experimental/ComboBox/docs/ComboBox.stories.tsx:188]: do we need these logs?

**Review by lloydaf — APPROVED** [review]

File: src/utils/hooks/useMergeRefs.ts (hunk 1 above)
Comment (AlanFnz) [reply src/utils/hooks/useMergeRefs.ts:4]: Yes, its functionally the same, Boolean would be in this case kind of a shorthand for filtering out falsy values. I can update it to use `ref => ref` for explicitness if that's preferred. Let me know!

File: src/components/experimental/ComboBox/docs/ComboBox.stories.tsx (hunk 2 above)
Comment (AlanFnz) [reply src/components/experimental/ComboBox/docs/ComboBox.stories.tsx:188]: Hmm not really, in the case of focus and blur, the change in the input state is evident. As for `handleGetValue` maybe we can remove it completely, the focus and blur actions are demonstration enough probably.

File: src/utils/hooks/useMergeRefs.ts (hunk 1 above)
Comment (lloydaf) [reply src/utils/hooks/useMergeRefs.ts:4]: I think it's fine the way it is.. thanks!

File: src/components/experimental/ComboBox/docs/ComboBox.stories.tsx (hunk 2 above)
Comment (lloydaf) [reply src/components/experimental/ComboBox/docs/ComboBox.stories.tsx:188]: Oh this is for stories.. nevermind

**Pull request merged**